cinema/urls.py

```diff
@@ -1,10 +1,45 @@
-from django.urls import path
+from django.urls import path, include
 
-from cinema.views import movie_list, movie_detail
+from rest_framework import routers
+
+from cinema.views import (
+    GenreList,
+    GenreDetail,
+    ActorList,
+    ActorDetail,
+    CinemaHallViewSet,
+    MovieViewSet,
+)
+
+
+router = routers.DefaultRouter()
+router.register("movies", MovieViewSet)
+
+cinema_hall_list = CinemaHallViewSet.as_view(actions={
+    "get": "list",
+    "post": "create"})
+cinema_hall_detail = CinemaHallViewSet.as_view(actions={
+    "get": "retrieve",
+    "put": "update",
+    "patch": "partial_update",
+    "delete": "destroy"})
 
 urlpatterns = [
-    path("movies/", movie_list, name="movie-list"),
-    path("movies/<int:pk>/", movie_detail, name="movie-detail"),
+    path("genres/", GenreList.as_view(), name="genres-list"),
+    path("genres/<int:pk>/", GenreDetail.as_view(), name="genres-detail"),
+    path("actors/", ActorList.as_view(), name="actors-list"),
+    path("actors/<int:pk>/", ActorDetail.as_view(), name="actors-detail"),
+    path(
+        "cinema_halls/",
+        cinema_hall_list,
+        name="cinema-hall-list"
+    ),
+    path(
+        "cinema_halls/<int:pk>/",
+        cinema_hall_detail,
+        name="cinema-hall-detail"
+    ),
+    path("", include(router.urls)),
```

Comment:
No need to include if basename is not provided. Instead use:
urlpatterns = [list of urls] + router.urls